src/sentry/api/helpers/group_index.py

```diff
@@ -207,3 +225,99 @@ def validate(self, attrs):
             raise serializers.ValidationError(
                 'Other attributes cannot be updated when discarding')
         return attrs
+
+
+def handle_discard(request, group_list, projects, user, view_cls):
+    for project in projects:
+        if not features.has('projects:discard-groups', project, actor=user):
+            return Response({'detail': ['You do not have that feature enabled']}, status=400)
+
+    # grouped by project_id
+    groups_to_delete = defaultdict(list)
+
+    for group in group_list:
+        with transaction.atomic():
+            try:
+                tombstone = GroupTombstone.objects.create(
+                    previous_group_id=group.id,
+                    actor_id=user.id if user else None,
+                    **{name: getattr(group, name) for name in TOMBSTONE_FIELDS_FROM_GROUP}
+                )
+            except IntegrityError:
+                # in this case, a tombstone has already been created
+                # for a group, so no hash updates are necessary
+                pass
+            else:
+                groups_to_delete[group.project_id].append(group)
+
+                GroupHash.objects.filter(
+                    group=group,
+                ).update(
+                    group=None,
+                    group_tombstone_id=tombstone.id,
+                )
+
+    for project in projects:
+        to_delete = groups_to_delete.get(project.id)
+        if to_delete:
```

Comment:
Nitpick: We could just call this since there's a check at the start of `delete_groups` that also checks this.